Data/Sys/Shaders/IntegerScaling.glsl

```diff
@@ -0,0 +1,25 @@
+void main()
+{
+	float4 c0 = Sample();
```

Comment:
Is this actually needed?  I'm pretty sure this will get overridden in either case below (apart from the alpha channel, but I don't think that's used?)

It might be cleaner and faster to use `float4 c0 = float4(0.0, 0.0, 0.0, 0.0);` and then replace

```glsl
if (GetCoordinates().x < x_left || x_right < GetCoordinates().x || GetCoordinates().y < y_top || y_bottom < GetCoordinates().y)
{
  // ... black ...
}
else
{
  // ... sample ...
}
```

with

```glsl
if (x_left <= GetCoordinates().x && GetCoordinates().x <= x_right && y_top <= GetCoordinates().y && GetCoordinates().y <= y_bottom)
{
  // ... sample ...
}
```

Reply:
Yeah that's much better

Reply:
I'm pretty sure the if-statement should be using `&&` instead of `||` now ([per De Morgan's laws](https://en.wikipedia.org/wiki/De_Morgan%27s_laws)), but I don't actually know what `SampleLocation` returns when the location is out of bounds.  If it returns black, then you probably don't need the if-statement at all, but otherwise I think you'll get incorrect results if the window size isn't right.

Reply:
Testing it showed, it will not always return black when out of range.